### ISSUE TITLE
Is there any filtering functionalities on triplet sampling?

### ISSUE THREAD

**Issue by knao124** [KevinMusgrave/pytorch-metric-learning#104]
I'd like to ask a question about how to implement sampling with this package in the following situations.

Now I have the following data.
    - samples with the label "A"
    - samples with the label "B".
    - samples with the label "C".
    - samples with the label "not A".

"not A" means "B or C, or an unknown label other than A~C".

The sampling I want to implement is when a sample labeled "A" is an anchor, only the sample labeled "not A" is used for the negative.
In other words, I don't want to use samples labeled "not A" as anchor.
I want to use them only as negative.

In this case, how should I implement it using pytorch-metric-learning?

**Comment by KevinMusgrave** [comment]
@knao124 The [TwoStreamMetricLoss(https://kevinmusgrave.github.io/pytorch-metric-learning/trainers/#twostreammetricloss) trainer is similar to what you're saying. It assumes your dataset returns two streams of data: the first stream is used for anchors, and the other stream is used for positives/negatives. You can see an example notebook [here](https://github.com/KevinMusgrave/pytorch-metric-learning/blob/master/examples/notebooks/TwoStreamMetricLoss.ipynb) and [on google colab](https://colab.research.google.com/github/KevinMusgrave/pytorch-metric-learning/blob/master/examples/notebooks/TwoStreamMetricLoss.ipynb).

However, it's a bit different from what you're saying, because it assumes that the dataloader will return pairs of anchors and positives at each iteration. It doesn't allow for there to be a negative stream.

What does your Dataset class look like? Does it already return the "A" and "Not A" samples, or is that what you're trying to figure out how to do?

**Comment by KevinMusgrave** [comment]
Closing for now. Please reopen if you have further questions.
